src/swap/utils/formatDecimals.ts

```diff
@@ -6,8 +8,54 @@ export function formatDecimals(
   inputInDecimals = true,
   decimals = 18,
 ): string {
+  // Input validation
+  if (typeof amount !== 'string' || amount.trim() === '') {
+    throw new Error('Invalid input: amount must be a non-empty string');
+  }
+  if (!Number.isInteger(decimals) || decimals < 0) {
+    throw new Error('Invalid input: decimals must be a non-negative integer');
+  }
+  if (!/^(?:0|[1-9]\d*)(?:\.\d+)?$/.test(amount)) {
+    throw new Error(
+      'Invalid input: amount must be a non-negative number string',
+    );
+  }
+
+  let result: string;
+
   if (inputInDecimals) {
-    return (Number(amount) / 10 ** decimals).toString();
+    // If input is already in decimals, convert to readable amount
+    result = toReadableAmount(amount, decimals);
+  } else {
+    // If input is not in decimals, convert from readable amount
+    result = fromReadableAmount(amount, decimals);
   }
-  return (Number(amount) * 10 ** decimals).toString();
+
+  return result;
+}
+
+export function fromReadableAmount(amount: string, decimals: number): string {
```

Comment:
We want every function to be his own file. This is something we mentioned in the past a few times, as it is a more verbose way to create a library, but in the long run makes it easier to mantain and test.